### PR TITLE
Polynomial Sum & Product using Linked Lists

### LINKED ISSUE #3560: Polynomial Arithmetics in Java
## 🚀 Feature

Implementation of polynomial Sum & multiplication using linked list in Java

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yep!

### PR DESCRIPTION
**Pull request by Suvraneel**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yep!

### Description

A dynamic implementation of Polynomial Sum & Product using Linked Lists.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

resolves #3560

### CODE REVIEW

**Comment by Suvraneel** [comment]
Hello maintainers!  
Kindly review this... Followed all suggestions from previous PRs     :)
@HarshCasper @bhav09 
@siddharth25pandey @ankitaggarwal23

File: Java/math/polynomial.java [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+Sample Run:
+  Enter number of terms in 1st Polynomial:        5
+  Enter 1st Polynomial as <coefficient exponent> :        10 6 5 4 3 3 4 1 7 0
+  Enter number of terms in 2nd Polynomial:        3
+  Enter 2nd Polynomial as <coefficient exponent> :        6 3 2 1 5 0
+  1st Polynomial =        		10x^6 + 5x^4 + 3x^3 + 4x^1 + 7
+  2nd Polynomial =        		6x^3 + 2x^1 + 5
+  Resultant Sum Polynomial =      10x^6 + 5x^4 + 9x^3 + 6x^1 + 12
+  Resultant Product Polynomial =  60x^9 + 50x^7 + 68x^6 + 10x^5 + 55x^4 + 57x^3 + 8x^2 + 34x^1 + 35
+*/
```
Comment (bhav09) [inline Java/math/polynomial.java:203]: Its perfect, just add a new line at EOF.

Comment (Suvraneel) [reply Java/math/polynomial.java:203]: Done,
- Thanks :)

**Comment by Suvraneel** [comment]
@bhav09  @HarshCasper  @anushkrishnav  @gat786 
Resolved = "\n at EOF"
Please review :)

File: Java/README.md [hunk 2]
```diff
@@ -167,7 +167,8 @@ _add list here_
 - [Palindrome Number Using Recursion](math/PalindromeRecursion.java)
 - [Strong Number Using Recursion](math/check_strong_number.java)
 - [Factorial Using Big Integer](math/Big_Integer_Factorial.java)
-- [Factorial Using BigInteger](FactorialUsingBigInteger.java)
+- [Factorial Using BigInteger](math/FactorialUsingBigInteger.java)
```
Comment (siddharth25pandey) [inline Java/README.md]: Remove this line

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@Suvraneel Minor change is required

**Comment by Suvraneel** [comment]
@siddharth25pandey @HarshCasper @bhav09 
Resolved :)

(P.S ~ That was actually because I was trying to fix a broken link in the readme of Java/math/FactorialusingBigInteger.java)

**Review by siddharth25pandey — APPROVED** [review]
@Suvraneel LGTM, able to merge

**Comment by siddharth25pandey** [comment]
@all-contributors please add @Suvraneel for his code

**Pull request merged**